# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmCodegenTypes.cpp
# Commit: d6506b5ee07b
# Full Hash: d6506b5ee07b5c65f889ce7c2214b0172f205898
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2021-09-16 09:43:03
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Introduce indirect stubs to optimize call_indirect. r=lth
#   
#   This patch introduces indirect stubs.
#   An indirect stub is a stub that takes care of any switching activities needed for call_indirect.
#   Before this patch, we have to conservatively assume that any call_indirect's target can be from a foreign instance,
# ==============================================================================

diff -r 80a164c1054e -r d6506b5ee07b js/src/wasm/WasmCodegenTypes.cpp
--- a/js/src/wasm/WasmCodegenTypes.cpp	Wed Sep 15 11:02:57 2021 +0000
+++ b/js/src/wasm/WasmCodegenTypes.cpp	Wed Sep 15 11:10:15 2021 +0000
@@ -139,7 +139,7 @@
   u.func.lineOrBytecode_ = 0;
   u.func.beginToUncheckedCallEntry_ = 0;
   u.func.beginToTierEntry_ = 0;
-  MOZ_ASSERT(isEntry());
+  MOZ_ASSERT(isEntry() || isIndirectStub());
   MOZ_ASSERT(begin_ <= end_);
 }
 